### PR TITLE
llvm: use -Bsymbolic-functions with -fno-semantic-interposition

### PR DESCRIPTION
**Pull request by klensy**
Should get some perf?

https://maskray.me/blog/2021-05-09-fno-semantic-interposition
https://maskray.me/blog/2021-05-16-elf-interposition-and-bsymbolic
https://bugs.python.org/issue38980

TODO: maybe -Bsymbolic-non-weak-functions ?

### CODE REVIEW

**Comment by klensy** [comment]
perf run please?

File: src/bootstrap/llvm.rs [hunk 1]
```diff
@@ -760,7 +762,9 @@ fn configure_cmake(
             if target.contains("apple") || target.contains("windows") {
                 ldflags.push_all("-static-libstdc++");
             } else {
-                ldflags.push_all("-Wl,-Bsymbolic -static-libstdc++");
+                ldflags.push_all(
+                    "-Wl,-Bsymbolic-functions -fno-semantic-interposition -static-libstdc++",
```
Comment (erikdesjardins) [inline src/bootstrap/llvm.rs:766]: `-Bsymbolic-functions` is weaker (it only applies to functions) than the existing `-Bsymbolic`. You should keep `-Bsymbolic` unless it causes issues with `-fno-semantic-interposition`

Comment (klensy) [reply src/bootstrap/llvm.rs]: Uh, results not nice, reverted back :-(

**Comment by klensy** [comment]
No difference, probably that flag already enabled somewhere down the flag chain.

**Comment by Mark-Simulacrum** [comment]
Closing since this didn't seem to pan out. Thanks!

**Pull request closed without merging**